### PR TITLE
Feat/create company bounded context

### LINKED ISSUE #157: Move Company Bounded Context

### PR DESCRIPTION
**Pull request by BenOpenOfficiel**
Ticket n° #157 

It's first implementation. We can improve it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /.directory
 /docker-compose.override.yml
 *.sql*
+/.idea
```
Comment (Dev-Int) [inline .gitignore:4]: I prefer not to have system folders to specify.

You can add it in your global git settings :wink:

File: Makefile [hunk 2]
```diff
@@ -60,6 +60,10 @@ cc: ## Clear the cache in dev env
 	@rm -rf server/var/cache/*
 	@$(EXEC) php php -d memory_limit=-1 server/bin/console cache:warmup
 
+cc-test: ## Clear the cache in dev env
+	@rm -rf server/var/cache/*
+	@$(EXEC) php php -d memory_limit=-1 server/bin/console cache:warmup --env=test
+
```
Comment (Dev-Int) [inline Makefile]: Perhaps look at duplicate with lines 150-151

File: Makefile [hunk 3]
```diff
@@ -80,13 +84,24 @@ bash-php: ## Open bash in php container
 bash-client: ## Open bash in client container
 	@$(EXEC) client bash
 
-
```
Comment (Dev-Int) [inline Makefile]: I prefer 2 new line between each section

File: server/composer.json [hunk 4]
```diff
@@ -26,6 +26,7 @@
         "jms/serializer-bundle": "^3.8",
         "lexik/jwt-authentication-bundle": "^2.10",
         "phpdocumentor/reflection-docblock": "^5.2",
+        "phpunit/phpunit": "^9.5",
```
Comment (Dev-Int) [inline server/composer.json]: Duplicate with "symfony/phpunit-bridge" in require-dev. We have to choice which one we want :wink:

File: server/config/routes/administration/company.yaml [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
   path: /
-  controller: Administration\Infrastructure\Company\Controller\PostCompanyController
+  controller: Company\UI\Controller\CreateCompany
   methods: POST
 
 admin_company_get:
   path: /{uuid}
-  controller: Administration\Infrastructure\Company\Controller\GetCompanyController
+  controller: Company\UI\Controller\GetCompany
   methods: GET
```
Comment (Dev-Int) [inline server/config/routes/administration/company.yaml]: Controller does not exist any-more.

File: server/src/Company/Domain/Exception/CompanyNotFoundException.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Company\Domain\Exception;
+
+use Core\Domain\Exception\ApiExceptionInterface;
+
+class CompanyNotFoundException extends \RuntimeException implements ApiExceptionInterface
+{
+    public function __construct()
+    {
+        parent::__construct('Company not found !!!');
```
Comment (Dev-Int) [inline server/src/Company/Domain/Exception/CompanyNotFoundException.php]: We forgot to remove the extra exclamation marks :yum:

File: server/src/Company/Infrastructure/Doctrine/Repository/CompanyRepository.php [hunk 7]
```diff
@@ -32,56 +37,57 @@ public function __construct(ManagerRegistry $registry)
...
-            ->where('c.companyName = :companyName')
-            ->setParameter('companyName', $companyName)
+            ->where('c.companyName = :name')
+            ->setParameter('name', $name)
             ->getQuery()
             ->getOneOrNullResult()
         ;
 
-        return null !== $statement;
+        return !(null === $statement);
```
Comment (Dev-Int) [inline server/src/Company/Infrastructure/Doctrine/Repository/CompanyRepository.php]: You think it's clearest ? :thinking:

File: server/src/Company/UI/Controller/CreateCompany.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+        parent::__construct($serializer);
+    }
+
+    /**
+     * @throws \JsonException
+     */
+    public function __invoke(Request $request): Response
+    {
+        /** @var CreateCompanyCommand $companyCommand */
```
Comment (Dev-Int) [inline server/src/Company/UI/Controller/CreateCompany.php:44]: PhpStorm prefer `assert()`. What do you about it ?

File: server/src/Core/Domain/Common/Model/VO/ResourceUuid.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Core\Domain\Common\Model\VO;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class ResourceUuid implements ResourceUuidInterface
```
Comment (Dev-Int) [inline server/src/Core/Domain/Common/Model/VO/ResourceUuid.php:19]: Did you simplify the Uuid management mechanism?

**Review by Dev-Int — CHANGES_REQUESTED** [review]

File: server/src/Company/Infrastructure/Doctrine/Repository/CompanyRepository.php [hunk 10]
```diff
@@ -32,56 +37,57 @@ public function __construct(ManagerRegistry $registry)
...
      */
-    final public function existsWithName(string $companyName): bool
+    final public function existsWithName(string $name): bool
     {
         $statement = $this->createQueryBuilder('c')
             ->select(['1'])
-            ->where('c.companyName = :companyName')
-            ->setParameter('companyName', $companyName)
+            ->where('c.companyName = :name')
+            ->setParameter('name', $name)
```
Comment (Dev-Int) [inline server/src/Company/Infrastructure/Doctrine/Repository/CompanyRepository.php]: You prefer `$name` to `$companyName` ? :thinking:

File: .gitignore (hunk 1 above)
Comment (BenOpenOfficiel) [reply .gitignore:4]: if you dont use it in your gitignore and work with developpers who dont set it you will find it in the repository

File: Makefile (hunk 3 above)
Comment (BenOpenOfficiel) [reply Makefile]: Ok

File: server/src/Core/Domain/Common/Model/VO/ResourceUuid.php (hunk 9 above)
Comment (BenOpenOfficiel) [reply server/src/Core/Domain/Common/Model/VO/ResourceUuid.php:19]: yes

File: server/config/routes/administration/company.yaml (hunk 5 above)
Comment (BenOpenOfficiel) [reply server/config/routes/administration/company.yaml]: The tests for this controller are missing

File: server/src/Company/UI/Controller/CreateCompany.php (hunk 8 above)
Comment (BenOpenOfficiel) [reply server/src/Company/UI/Controller/CreateCompany.php:44]: I prefer explicite documentation

**Review by Dev-Int — APPROVED** [review]

**Pull request merged**